src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,110 @@
-export const Pagination = () => {};
+/* eslint-disable react/jsx-indent */
+import { FC } from 'react';
+
+interface Props {
+  total: number,
+  itemsPerPage: number,
+  currentPage: number,
+  setCurrentPage: (n: number) => void,
+  onPageChange: (page: number) => void,
+  items: string[],
+}
+
+export const Pagination: FC<Props> = (
+  {
+    total,
+    itemsPerPage,
+    currentPage = 1,
+    setCurrentPage,
+    onPageChange,
+    items,
+  },
+) => {
+  const pages: number[] = [];
+  const maxPages = Math.ceil(total / itemsPerPage);
+
+  for (let i = 1; i <= maxPages; i += 1) {
+    pages.push(i);
+  }
+
+  const handlePageChange = (page: number): void => {
+    onPageChange(page);
+    setCurrentPage(page);
+  };
+
+  const preparedItems = items.slice((currentPage - 1) * itemsPerPage,
+    currentPage * itemsPerPage);
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={currentPage === 1 ? 'page-item disabled' : 'page-item'}>
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 1}
+            onClick={() => {
+              const newPage = currentPage - 1;
```

Comment:
redundant variable, create in the top level above `maxPages` and reuse, check in all places